asmcomp/debug/dwarf_high/operator_builder.ml

```diff
@@ -0,0 +1,201 @@
+  | 7 -> DW_op_reg7
+  | 8 -> DW_op_reg8
+  | 9 -> DW_op_reg9
+  | 10 -> DW_op_reg10
+  | 11 -> DW_op_reg11
+  | 12 -> DW_op_reg12
+  | 13 -> DW_op_reg13
+  | 14 -> DW_op_reg14
+  | 15 -> DW_op_reg15
+  | 16 -> DW_op_reg16
+  | 17 -> DW_op_reg17
+  | 18 -> DW_op_reg18
+  | 19 -> DW_op_reg19
+  | 20 -> DW_op_reg20
+  | 21 -> DW_op_reg21
+  | 22 -> DW_op_reg22
+  | 23 -> DW_op_reg23
+  | 24 -> DW_op_reg24
+  | 25 -> DW_op_reg25
+  | 26 -> DW_op_reg26
+  | 27 -> DW_op_reg27
+  | 28 -> DW_op_reg28
+  | 29 -> DW_op_reg29
+  | 30 -> DW_op_reg30
+  | 31 -> DW_op_reg31
+  | _ -> DW_op_regx { reg_number = dwarf_reg_number; }
+
+let dw_op_bregx ~dwarf_reg_number ~offset_in_bytes : O.t =
+  match dwarf_reg_number with
+  | 0 -> DW_op_breg0 { offset_in_bytes; }
+  | 1 -> DW_op_breg1 { offset_in_bytes; }
+  | 2 -> DW_op_breg2 { offset_in_bytes; }
+  | 3 -> DW_op_breg3 { offset_in_bytes; }
+  | 4 -> DW_op_breg4 { offset_in_bytes; }
+  | 5 -> DW_op_breg5 { offset_in_bytes; }
+  | 6 -> DW_op_breg6 { offset_in_bytes; }
+  | 7 -> DW_op_breg7 { offset_in_bytes; }
+  | 8 -> DW_op_breg8 { offset_in_bytes; }
+  | 9 -> DW_op_breg9 { offset_in_bytes; }
+  | 10 -> DW_op_breg10 { offset_in_bytes; }
+  | 11 -> DW_op_breg11 { offset_in_bytes; }
+  | 12 -> DW_op_breg12 { offset_in_bytes; }
+  | 13 -> DW_op_breg13 { offset_in_bytes; }
+  | 14 -> DW_op_breg14 { offset_in_bytes; }
+  | 15 -> DW_op_breg15 { offset_in_bytes; }
+  | 16 -> DW_op_breg16 { offset_in_bytes; }
+  | 17 -> DW_op_breg17 { offset_in_bytes; }
+  | 18 -> DW_op_breg18 { offset_in_bytes; }
+  | 19 -> DW_op_breg19 { offset_in_bytes; }
+  | 20 -> DW_op_breg20 { offset_in_bytes; }
+  | 21 -> DW_op_breg21 { offset_in_bytes; }
+  | 22 -> DW_op_breg22 { offset_in_bytes; }
+  | 23 -> DW_op_breg23 { offset_in_bytes; }
+  | 24 -> DW_op_breg24 { offset_in_bytes; }
+  | 25 -> DW_op_breg25 { offset_in_bytes; }
+  | 26 -> DW_op_breg26 { offset_in_bytes; }
+  | 27 -> DW_op_breg27 { offset_in_bytes; }
+  | 28 -> DW_op_breg28 { offset_in_bytes; }
+  | 29 -> DW_op_breg29 { offset_in_bytes; }
+  | 30 -> DW_op_breg30 { offset_in_bytes; }
+  | 31 -> DW_op_breg31 { offset_in_bytes; }
+  | _ -> DW_op_bregx { reg_number = dwarf_reg_number; offset_in_bytes; }
+
+let register_as_lvalue ~dwarf_reg_number =
+  dw_op_regx ~dwarf_reg_number
+
+let contents_of_register ~dwarf_reg_number =
+  dw_op_bregx ~dwarf_reg_number ~offset_in_bytes:Targetint.zero
+
+let address_of_stack_slot ~offset_in_bytes =
+  (* Note that this isn't target-dependent.  The target dependent part
+     is the calculation of the [offset_in_bytes]. *)
+  [ O.DW_op_call_frame_cfa;
+    O.DW_op_consts offset_in_bytes;
+    O.DW_op_minus;
+  ]
+
+let contents_of_stack_slot ~offset_in_bytes =
+  (* Same comment as per [address_of_stack_slot]. *)
+  [ O.DW_op_call_frame_cfa;
+    O.DW_op_consts offset_in_bytes;
+    O.DW_op_minus;
+    O.DW_op_deref;
+  ]
+
+let value_of_symbol ~symbol : O.t = DW_op_addr symbol
+
+let signed_int_const i : O.t = DW_op_consts i
```

Comment:
Given that there are now the different size variant of DW_op_const maybe you could chose the one that fits best to safe some space.